lib/ansible/modules/cloud/vmware/vmware_guest_network.py

```diff
@@ -0,0 +1,453 @@
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: vmware_guest_network
+short_description: Manage network adapters of specified virtual machine in given vCenter infrastructure
+description:
+    - This module is used to add, reconfigure, remove network adapter of given virtual machine.
+    - All parameters and VMware object names are case sensitive.
+version_added: '2.8'
+author:
+    - Diane Wang (@Tomorrow9) <dianew@vmware.com>
+notes:
+    - Tested on vSphere 6.0, 6.5 and 6.7
+requirements:
+    - "python >= 2.6"
+    - PyVmomi
+options:
+   name:
+     description:
+     - Name of the virtual machine.
+     - This is a required parameter, if parameter C(uuid) is not supplied.
+   uuid:
+     description:
+     - UUID of the instance to gather facts if known, this is VMware's unique identifier.
+     - This is a required parameter, if parameter C(name) is not supplied.
+   folder:
+     description:
+     - Destination folder, absolute or relative path to find an existing guest.
+     - This is a required parameter, only if multiple VMs are found with same name.
+     - The folder should include the datacenter. ESXi server's datacenter is ha-datacenter.
+     - 'Examples:'
+     - '   folder: /ha-datacenter/vm'
+     - '   folder: ha-datacenter/vm'
+     - '   folder: /datacenter1/vm'
+     - '   folder: datacenter1/vm'
+     - '   folder: /datacenter1/vm/folder1'
+     - '   folder: datacenter1/vm/folder1'
+     - '   folder: /folder1/datacenter1/vm'
+     - '   folder: folder1/datacenter1/vm'
+     - '   folder: /folder1/datacenter1/vm/folder2'
+   cluster:
+     description:
+     - The name of cluster where the virtual machine will run.
+     - This is a required parameter, if C(esxi_hostname) is not set.
+     - C(esxi_hostname) and C(cluster) are mutually exclusive parameters.
+   esxi_hostname:
+     description:
+     - The ESXi hostname where the virtual machine will run.
+     - This is a required parameter, if C(cluster) is not set.
+     - C(esxi_hostname) and C(cluster) are mutually exclusive parameters.
+   datacenter:
+     default: ha-datacenter
+     description:
+     - The datacenter name to which virtual machine belongs to.
+   gather_network_facts:
+     description:
+     - If set to True, return settings of all network adapters, other parameters are ignored.
+     - If set to False, will add, reconfigure or remove network adapters according to the parameters in C(networks).
+     type: bool
+     default: False
+   networks:
+     description:
+     - A list of network adapters.
+     - C(label) or C(device_type) is required to reconfigure or remove an existing network adapter.
+     - 'If there are multiple network adapters with the same C(device_type), you should set C(label),
+        or will use the first matched network adapter.'
```

Comment:
Sounds a strange behavior to me to apply the change only on the first matched NIC when the definition does not indicate a single object. Either apply change on all matched NICs or fails sounds more logical to me.

Reply:
Changed to do reconfigure or remove on the all matched network adapters with the "device_type" if no "mac" or "label" specified or no matched. 
Add "manual_mac" parameter to reconfigure network adapter's MAC address. Thanks.